### PR TITLE
Add github actions.

### PR DESCRIPTION
**Pull request by terabytesoftw**
Hi, add tests for github actions, running for code coverage both xdebug, and pcov, like tests in travis.yml, currently tests fail on php 7.3, 7.4 and 8.0 on ubuntu-lastest and on windows-lastets they fail the tests for the cli suite, then just add the tests for ubuntu-latest, if you correct the tests you can add everything in one workflow.

fail tests ubuntu-latest 7.3, 7.4, 8.0: it is configured to allow fail for now.
https://github.com/terabytesoftw/Codeception/actions/runs/417546115

fail test windows-latest:
https://github.com/terabytesoftw/Codeception/runs/1542676219?check_suite_focus=true

### CODE REVIEW

**Comment by samdark** [comment]
`travis.yml` should be removed.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    continue-on-error: ${{ matrix.experimental }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        php: ["5.6", "7.0", "7.1", "7.2"]
+        coverage: [xdebug]
+        experimental: [false]
+        include:
```
Comment (samdark) [inline .github/workflows/build.yml:26]: Why testing 7.3+ against pcov only? XDebug 3 is released and should work well with `XDEBUG_MODE=coverage` in environment vairables.

**Comment by samdark** [comment]
`.travis.yml` should be removed.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      SYMFONY_DEPRECATIONS_HELPER: weak
+
+    runs-on: ${{ matrix.os }}
+
+    continue-on-error: ${{ matrix.experimental }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
```
Comment (samdark) [inline .github/workflows/build.yml:22]: Can it run on Windows? If so, `appveyor.yml` could be removed.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
+on:
+  - pull_request
```
Comment (samdark) [inline .github/workflows/build.yml:2]: Originally `gh-pages` branch was excluded from triggering tests. I think it's doable with actions as well.

**Comment by samdark** [comment]
Readme badges should be updated.

**Comment by samdark** [comment]
Added "critical" lavel since TavisCI isn't giving enough credits for OpenSource anymore. Therefore, currently Codeception is not properly tested.

File: .github/workflows/build.yml (hunk 2 above)
Comment (terabytesoftw) [reply .github/workflows/build.yml:22]: done.

File: .github/workflows/build.yml (hunk 3 above)
Comment (terabytesoftw) [reply .github/workflows/build.yml]: done.

File: .github/workflows/build.yml (hunk 1 above)
Comment (terabytesoftw) [reply .github/workflows/build.yml]: done.

**Comment by terabytesoftw** [comment]
- Add the tests for `windows-latest` `php 7.1`, which is the ones that run in appveyor, test from windows `5.6` to `8.0` but fail `5.6`, `7.3`, `7.4`, `8.0`.
- All tests run with `xdebug`.
- Add allow failure in `php 8.0`.
- In the readme they have two badges, one for `general`, and one for `windows`, if you want the same, just tell me and i will divide the two workflows separately.

**Comment by samdark** [comment]
> Add allow failure in php 8.0.

I think that's a bad idea.

**Comment by samdark** [comment]
> In the readme they have two badges, one for general, and one for windows, if you want the same, just tell me and i will divide the two workflows separately.

A single badge would do.

**Comment by samdark** [comment]
> All tests run with xdebug.

Why do we still use pcov then?

**Comment by ctrl-f5** [comment]
Please also check my PR, it has some fixes for the failing tests: https://github.com/Codeception/Codeception/pull/6050

**Comment by terabytesoftw** [comment]
> > Add allow failure in php 8.0.
> 
> I think that's a bad idea.

- Remove allow failure php 8.0, add badge.

**Comment by Naktibalda** [comment]
Try to restrict phpunit to `<9.5` and see if the builds pass.
PHPUnit 9.5 compatibility can be fixed later.

**Comment by terabytesoftw** [comment]
> Try to restrict phpunit to `<9.5` and see if the builds pass.
> PHPUnit 9.5 compatibility can be fixed later.

it worked, add checks for windows-latest from 7.0 - 8.0, 5.6 fails.

**Review by samdark — APPROVED** [review]
Awesome!

**Comment by samdark** [comment]
@Naktibalda anything missing?

**Review by Naktibalda — APPROVED** [review]
Looks great.

**Comment by DavertMik** [comment]
Thanks for all the work!

**Comment by eXorus** [comment]
Amazing team work, good job!

**Comment by Naktibalda** [comment]
I "fixed" failing test https://github.com/Codeception/Codeception/pull/6054/files

**Pull request merged**